### PR TITLE
Add v2.10.2+spring2024 to release version 

### PR DESCRIPTION
**Pull request by hiltonlima**
- Add v2.10.2 release to release history
- Update links

### CODE REVIEW

**Review by rquidute — APPROVED** [review]

**Review by antonio-amjr — APPROVED** [review]

File: docs/Matter_TH_User_Guide/Matter_TH_User_Guide.adoc [hunk 1]
```diff
@@ -49,8 +49,9 @@ endif::[]
...
-| Matter 1.4       | v2.11+fall2024 | N/A                                                                                         | https://groups.csa-iot.org/wg/members-all/document/folder/4120[Causeway Link]  | f2e5de7      | To install, use the tag "v2.11+fall2024" in the instructions of <<fresh_install>>
+| Matter 1.3       | v2.10.1+spring2024 | N/A                                                                                         | https://groups.csa-iot.org/wg/members-all/document/folder/3314[Causeway Link] | fb2c0e5      | To install, use the tag "v2.10.1+spring2024" in the instructions of <<fresh_install>>
+| Matter 1.3       | v2.10.2+spring2024 | N/A                                                                                         | https://groups.csa-iot.org/wg/members-all/document/folder/3314[Causeway Link] | 50bcad1      | To install, use the tag "v2.10.2+spring2024" in the instructions of <<fresh_install>>
```
Comment (ccruzagralopes) [inline docs/Matter_TH_User_Guide/Matter_TH_User_Guide.adoc:53]: Is Matter 1.3 duplicated?

Comment (hiltonlima) [reply docs/Matter_TH_User_Guide/Matter_TH_User_Guide.adoc:53]: `v2.10.1+spring2024` and  `v2.10.2+spring2024`

Comment (hiltonlima) [reply docs/Matter_TH_User_Guide/Matter_TH_User_Guide.adoc:53]: `v2.10+spring2024`  too

Comment (ccruzagralopes) [reply docs/Matter_TH_User_Guide/Matter_TH_User_Guide.adoc:53]: Oh, I see

**Review by ccruzagralopes — APPROVED** [review]

**Pull request merged**